src/annotator/integrations/vitalsource.ts

```diff
@@ -436,7 +505,23 @@ export class VitalSourceContentIntegration
       return selectors;
     }
 
-    const pageInfo = await this._bookElement.getCurrentPage();
+    const [pageInfo, toc] = await Promise.all([
+      this._bookElement.getCurrentPage(),
+      this._bookElement.getTOC(),
+    ]);
```

Comment:
And if either of these rejects? e.g. If I make our faked Mosaic-book-element's `getTOC` throw, the rejected Promise is unhandled.

Reply:
> And if either of these rejects?

Then creating an annotation will fail. If the network request that provides the data for `getTOC` fails for any reason, then the `getTOC` Promise still resolves but with `{ status: false }`. If there is a scenario where `getTOC` rejects, I'd like it to do so in a noticeable way, so we can find out about it and deal with it as appropriate.